sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -243,28 +246,55 @@ class NodeHttpClient implements HttpClient {
   }
 
   private getOrCreateAgent(request: PipelineRequest, isInsecure: boolean): http.Agent {
-    if (!request.disableKeepAlive) {
-      if (isInsecure) {
-        if (!this.httpKeepAliveAgent) {
-          this.httpKeepAliveAgent = new http.Agent({
-            keepAlive: true,
-          });
-        }
+    const disableKeepAlive = request.disableKeepAlive;
 
-        return this.httpKeepAliveAgent;
-      } else {
-        if (!this.httpsKeepAliveAgent) {
-          this.httpsKeepAliveAgent = new https.Agent({
-            keepAlive: true,
-          });
-        }
+    // Handle Insecure requests first
+    if (isInsecure) {
+      if (disableKeepAlive) {
+        // keepAlive:false is the default so we don't need a custom Agent
+        return http.globalAgent;
+      }
 
-        return this.httpsKeepAliveAgent;
+      // Get the cached agent if exists
+      let agent = this.cachedHttpAgent;
+      if (!agent) {
+        // If there is no cached agent create a new one and cache it.
+        agent = new http.Agent({ keepAlive: true });
+        this.cachedHttpAgent;
       }
```

Comment:
Looks like this didn't cache the new agent, easy enough to fix since we don't really need the temporary variable:
```suggestion
      if (!this.cachedHttpAgent) {
        // If there is no cached agent create a new one and cache it.
        this.cachedHttpAgent = new http.Agent({ keepAlive: true });
      }
      return this.cachedHttpAgent;
```

Reply:
🤦‍♂️ Thanks for noticing!